yt_dlp/extractor/sbscokr.py

```diff
@@ -0,0 +1,200 @@
+                'series': ('info', 'program', 'programtitle'),
+                'age_limit': ('info', 'targetage', {int_or_none}),
+                'release_timestamp': ('info', 'broaddate', {parse_iso8601}),
+                'thumbnail': ('source', 'thumbnail', 'origin', {url_or_none}),
+            }), get_all=False),
+            'formats': formats,
+            'subtitles': subtitles,
+        }
+
+
+class SBSCoKrAllvodProgramIE(InfoExtractor):
+    IE_NAME = 'sbs.co.kr:allvod_program'
+    _VALID_URL = r'https?://allvod\.sbs\.co\.kr/allvod/vod(?:Free)?ProgramDetail\.do\?(?:[^#]+&)?pgmId=(?P<id>P?[0-9]+)'
+
+    _TESTS = [{
+        'url': 'https://allvod.sbs.co.kr/allvod/vodFreeProgramDetail.do?type=legend&pgmId=22000010159&listOrder=vodCntAsc',
+        'info_dict': {
+            '_type': 'playlist',
+            'id': '22000010159',
+        },
+        'playlist_count': 18,
+    }, {
+        'url': 'https://allvod.sbs.co.kr/allvod/vodProgramDetail.do?pgmId=P460810577',
+        'info_dict': {
+            '_type': 'playlist',
+            'id': 'P460810577',
+        },
+        'playlist_count': 13,
+    }]
+
+    def _real_extract(self, url):
+        program_id = self._match_id(url)
+
+        details = self._download_json(
+            'https://allvod.sbs.co.kr/allvod/vodProgramDetail/vodProgramDetailAjax.do',
+            program_id, note='Downloading program details',
+            query={
+                'pgmId': program_id,
+                'currentCount': '10000',
+            })
+
+        return self.playlist_result(
+            [self.url_result(
+                'https://allvod.sbs.co.kr/allvod/vodEndPage.do?mdaId=' + video_id, SBSCoKrIE
+            ) for video_id in traverse_obj(details, ('list', ..., 'mdaId'), default=[])], program_id)
+
+
+class SBSCoKrProgramsVodIE(InfoExtractor):
+    IE_NAME = 'sbs.co.kr:programs_vod'
+    _VALID_URL = r'https?://programs\.sbs\.co\.kr/(?:enter|drama|culture|sports|plus|mtv)/(?P<id>[a-z0-9]+)/vods'
+
+    _TESTS = [{
+        'url': 'https://programs.sbs.co.kr/culture/morningwide/vods/65007',
+        'info_dict': {
+            '_type': 'playlist',
+            'id': '00000210215',
+        },
+        'playlist_mincount': 9782,
+    }, {
+        'url': 'https://programs.sbs.co.kr/enter/dongsang2/vods/52006',
+        'info_dict': {
+            '_type': 'playlist',
+            'id': '22000010476',
+        },
+        'playlist_mincount': 312,
+    }]
+
+    def _real_extract(self, url):
+        program_slug = self._match_id(url)
+
+        menu_data = self._download_json(
+            f'https://static.apis.sbs.co.kr/program-api/1.0/menu/{program_slug}', program_slug,
+            note='Downloading program menu data')
+
+        return self.url_result(
+            'https://allvod.sbs.co.kr/allvod/vodProgramDetail.do?pgmId=' + traverse_obj(
+                menu_data, ('program', 'programid')), SBSCoKrAllvodProgramIE)
```

Comment:
IMO if/when this breaks, a keyerror would be cleaner than returning a url_result with a malformed url
```suggestion
        program_id = self._download_json(
            f'https://static.apis.sbs.co.kr/program-api/1.0/menu/{program_slug}', program_slug,
            note='Downloading program menu data')['program']['programid']

        return self.url_result(
            f'https://allvod.sbs.co.kr/allvod/vodProgramDetail.do?pgmId={program_id}', SBSCoKrAllvodProgramIE)
```